### PR TITLE
Add CI for compile check

### LINKED ISSUE #3: CI research and implementation
Research and implement a proper CI configuration for this project.

> Hint: look into how seaport does this.

### PR DESCRIPTION
**Pull request by fkrause98**
Closes #3

### CODE REVIEW

File: ExecutionHelper/hardhat.config.ts [hunk 1]
```diff
@@ -26,8 +26,8 @@ const config: HardhatUserConfig = {
       zksync: true,
     },
     inMemoryNode: {
-      url: "http://localhost:8011",
-      ethNetwork: "", // in-memory node doesn't support eth node; removing this line will cause an error
+      url: "http://127.0.0.1:8011",
+      ethNetwork: null, // in-memory node doesn't support eth node; removing this line will cause an error
```
Comment (jpcenteno) [inline ExecutionHelper/hardhat.config.ts:30]: The typescript LSP told me to replace `null` with `undefined`. We should check wether that works.

File: docker-compose.yml
Comment (jpcenteno) [inline docker-compose.yml:1]: This file is unused. It should be deleted.

File: .github/workflows/compile-check.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Setup project dependencies 
+        run: |
+          yarn global add node-gyp@10.0.1 && yarn install
+
+      - name: Compile and deploy ExecutionHelper.
+        working-directory: ./ExecutionHelper
+        run: |
+          yarn install
+          yarn hardhat compile
+          yarn hardhat deploy-zksync --script deploy.ts --network inMemoryNode
```
Comment (jpcenteno) [inline .github/workflows/compile-check.yml:28]: We should split these steps and use our makefile. Maybe we can do that in another PR

**Review by jpcenteno — CHANGES_REQUESTED** [review]

File: ExecutionHelper/hardhat.config.ts (hunk 1 above)
Comment (fkrause98) [reply ExecutionHelper/hardhat.config.ts]: I think this should actually be an empty string, I'll put that.

File: docker-compose.yml
Comment (fkrause98) [reply docker-compose.yml:1]: Done!

File: .github/workflows/compile-check.yml (hunk 2 above)
Comment (fkrause98) [reply .github/workflows/compile-check.yml:28]: Yeah, I think we can do this right here, let me check

File: .github/workflows/compile-check.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: actions/checkout@v3
+
+      - name: Set up Node.js 18.18
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.18
+
+      - name: Setup test-node 
+        run: |
+          make start-node-from-release-binary &
```
[the remark this answers is not in the document]
Comment (fkrause98) [reply .github/workflows/compile-check.yml]: Closing this since we're using the gh action now.

File: .github/workflows/compile-check.yml (hunk 2 above)
Comment (fkrause98) [reply .github/workflows/compile-check.yml]: MMm, that did not turned out as I was expecting it, lets check it on another PR, unless you have a clear idea on how to approach this?

**Review by daphneherlambda — APPROVED** [review]
LGTM!!

**Review by jpcenteno — COMMENTED** [review]
LGTM !

**Review by jpcenteno — APPROVED** [review]
